### DISCUSSION TITLE
Replacing network loaded icons with data ones

### DISCUSSION THREAD

**Discussion by rezad1393** [Purfview/IMDb-Scout-Mod#51] [Ideas]
can you replace more network loaded icons with data: ones
so that they loaded faster for internet of mortal ones like me?

**Comment by Purfview** [comment]
If you meant that all of them, then no.

**Because of cons:**
Lots of work.
Lots of maintenance.
Size of the script.

**Now I optimize and add them to the script if:**
Icon doesn't work or doesn't show for other reasons.
Icon's size is high, more than ~25kB.
Icon's quality is very low and there is some logo in higher quality on site.

We are talking about the downloading sites, icons takes what, maybe ~one MB all of them, and they are cached by a browser.
From time to time I check what icons needs to be updated or moved to the script.
I can add a setting to disable icons in settings. (EDIT: added to v10.6)

**Reply by rezad1393** [reply]
I put this issue here in discussion because I know it is not felt by all users, just by some.
for example for me some of those websites are filtered and I user vpn and tor to connect to them.
and because of the many of them are behind a MITM... I mean CDN named cloudflare the icon dont load with 403 errors.
like opensubtitle for me.

but you are right that it makes the script bigger (though I dont think that is that bad because script is local and less of a burden  than network loaded scripts.

the maintenance part I think is not bad because it is one time done deal.

about caching:
I, and many other, use addons that disable cache (firefox by default did that for https till very recently) or make firefox remove cache after the tab (or all tab of that domain, like by this addon "cookie-autodelete") are closed.

but still if it is too much for all can I request some here that I use most and you can add them?

I am trying to minimize the diff between my modified script based on you r script and I am down to the custom sites only and changing the icons on my script make the diff that I apply manually after your every update way more cumbersome.

**Reply by Purfview** [reply]
> ...I user vpn ...CDN named cloudflare the icon dont load with 403 errors. like opensubtitle for me.
  
Maybe because you use VPN with shared IP and CF is using stricter settings on those. I'll move opensubtitle to the script.

> ...it makes the script bigger ...less of a burden than network loaded scripts.

Script hosting sites have size limit.

> the maintenance part I think is not bad because it is one time done deal.

Not one time. Because sites can change design and icon. Plus external icons make easier to spot some things.

> I, and many other, use addons that disable cache...

Then you shouldn't worry about network traffic at all.

> I am trying to minimize the diff between my modified script based on you r script and I am down to the custom sites only and changing the icons on my script make the diff that I apply manually after your every update way more cumbersome.

You can set updating interval to week, month or more.
